tests/test_client.py

```diff
@@ -534,70 +533,72 @@ def test_extra_headers(
     """
     client.session.headers = session_headers
 
-    mock_response = Mock(spec=Response)
-    mock_response.headers = {"content-type": "text"}
-    mock_response.text = "some error"
-    mock_response.status_code = 400
-    request_mock.return_value = mock_response
+    mock_response = responses.post(
+        url=client._url_from_endpoint("multiscan", "v1"),
+        content_type="text/plain",
+        body="some error",
+        status=400,
+        match=[matchers.header_matcher(extra_headers)] if extra_headers else [],
+    )
 
     client.multi_content_scan(
         [{"filename": FILENAME, "document": DOCUMENT}],
         extra_headers=extra_headers,
     )
-    assert request_mock.called
-    _, kwargs = request_mock.call_args
-    assert expected_headers == kwargs["headers"]
```

Comment:
We don't test anymore if the extra_headers are used in the request.

Reply:
It's now tested by the `match=[...` part of the `responses.post()` call: if the correct headers are not passed then the call to `client.multi_content_scan()` fails because `responses` won't find a mocked response matching the request.